tests/benchmarks/peripheral_stress_test/src/i2s_thread.c

```diff
@@ -0,0 +1,201 @@
+};
+
+/* The data_r represent a sine wave with double the frequency of data_l */
+static int16_t data_r[SAMPLE_NO] = {
+	 12539,  23169,  30272,  32767,  30272,  23169,  12539,      0, -12540,
+	-23170, -30273, -32767, -30273, -23170, -12540,     -1,  12539,  23169,
+	 30272,  32767,  30272,  23169,  12539,      0, -12540, -23170, -30273,
+	-32767, -30273, -23170, -12540,     -1,
+};
+
+#define BLOCK_SIZE (2 * sizeof(data_l))
+
+K_MEM_SLAB_DEFINE(rx_mem_slab, BLOCK_SIZE, NUM_RX_BLOCKS, 32);
+K_MEM_SLAB_DEFINE(tx_mem_slab, BLOCK_SIZE, NUM_TX_BLOCKS, 32);
+
+
+static void fill_buf(int16_t *tx_block, int att)
+{
+	for (int i = 0; i < SAMPLE_NO; i++) {
+		tx_block[2 * i] = data_l[i] >> att;
+		tx_block[2 * i + 1] = data_r[i] >> att;
+	}
+}
+
+int configure_stream(const struct device *dev_i2s, enum i2s_dir dir)
+{
+	int ret;
+	struct i2s_config i2s_cfg;
+
+	i2s_cfg.word_size = 16U;
+	i2s_cfg.channels = 2U;
+	i2s_cfg.format = I2S_FMT_DATA_FORMAT_I2S;
+	i2s_cfg.frame_clk_freq = FRAME_CLK_FREQ;
+	i2s_cfg.block_size = BLOCK_SIZE;
+	i2s_cfg.timeout = TIMEOUT;
+
+	if (dir == I2S_DIR_TX) {
+		/* Configure the Transmit port as Master */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER
+				| I2S_OPT_BIT_CLK_MASTER;
+	} else if (dir == I2S_DIR_RX) {
+		/* Configure the Receive port as Slave */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_SLAVE
+				| I2S_OPT_BIT_CLK_SLAVE;
+	} else { /* dir == I2S_DIR_BOTH */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER
+				| I2S_OPT_BIT_CLK_MASTER;
+	}
+
+	if (dir == I2S_DIR_TX || dir == I2S_DIR_BOTH) {
+		i2s_cfg.mem_slab = &tx_mem_slab;
+		ret = i2s_configure(dev_i2s, I2S_DIR_TX, &i2s_cfg);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure I2S TX stream (%d)", ret);
+			return -1;
+		}
+	}
+
+	if (dir == I2S_DIR_RX || dir == I2S_DIR_BOTH) {
+		i2s_cfg.mem_slab = &rx_mem_slab;
+		ret = i2s_configure(dev_i2s, I2S_DIR_RX, &i2s_cfg);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure I2S RX stream (%d)", ret);
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+static void tx_block_write(const struct device *dev_i2s, int att, int err)
+{
+	char tx_block[BLOCK_SIZE];
+	int ret;
+
+	fill_buf((uint16_t *)tx_block, att);
+	ret = i2s_buf_write(dev_i2s, tx_block, BLOCK_SIZE);
+	if (ret != err) {
+		LOG_ERR("i2s_buf_write failed (%d)", ret);
+	}
+}
+
+static void rx_block_read(const struct device *dev_i2s, int att)
+{
+	char rx_block[BLOCK_SIZE];	/* Data is received but left unalyzed */
+	size_t rx_size;
+	int ret;
+
+	ret = i2s_buf_read(dev_i2s, rx_block, &rx_size);
+	if (ret < 0 || rx_size != BLOCK_SIZE) {
+		LOG_ERR("i2s_buf_read() failed (%d)", ret);
+	} else {
+		LOG_DBG("i2s_buf_read() received %d", rx_size);
+	}
+}
+
+/* I2S thread */
+static void i2s_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	int ret;
+
+	atomic_inc(&started_threads);
+
+	if (!device_is_ready(dev_i2s)) {
+		LOG_ERR("Device %s is not ready.", dev_i2s->name);
+		atomic_inc(&completed_threads);
```

Comment:
Not needed here? missing increment will allow to detect that something has failed

Reply:
Failure detected based on LOG_INF("I2S thread has completed");